account_credit_control/models/credit_control_communication.py

```diff
@@ -180,60 +177,23 @@ def _generate_comm_from_credit_lines(self, lines):
         """
         datas = self._aggregate_credit_lines(lines)
         comms = self.create(datas)
+        comms._onchange_partner_id()
         return comms
 
     @api.multi
-    @api.returns('mail.mail')
     def _generate_emails(self):
         """ Generate email message using template related to level """
-        emails = self.env['mail.mail']
-        required_fields = [
-            'subject',
-            'body_html',
-            'email_from',
-            'email_to',
-        ]
         for comm in self:
-            template = comm.current_policy_level.email_template_id
-            email_values = template.generate_email(comm.id)
-            email_values['message_type'] = 'email'
-            # model is Transient record (self) removed periodically so no point
-            # of storing res_id
-            email_values.pop('model', None)
-            email_values.pop('res_id', None)
-            # Remove when mail.template returns correct format attachments
-            attachment_list = email_values.pop('attachments', [])
-            email = emails.create(email_values)
-
-            state = 'sent'
-            # The mail will not be send, however it will be in the pool, in an
-            # error state. So we create it, link it with
-            # the credit control line
-            # and put this latter in a `email_error` state we not that we have
-            # a problem with the email
-            if not all(email_values.get(field) for field in required_fields):
-                state = 'email_error'
-            comm.credit_control_line_ids.write({
-                'mail_message_id': email.id,
-                'state': state,
-            })
-            email.attachment_ids = [(0, 0, {
-                'name': att[0],
-                'datas': att[1],
-                'datas_fname': att[0],
-                'res_model': 'mail.mail',
-                'res_id': email.id,
-                'type': 'binary',
-            }) for att in attachment_list]
-            emails |= email
-        return emails
+            comm.message_post_with_template(
+                comm.policy_level_id.email_template_id.id,
+                composition_mode="mass_post",
```

Comment:
Is this the suitable mode for having the thread available for answers?

Reply:
Yes, it's like posting a normal message, but it uses the mail queue instead of posting directly.